src/webauth/agent.js

```diff
@@ -15,15 +19,21 @@ export default class Agent {
     return new Promise((resolve, reject) => {
       const urlHandler = event => {
         Authcore.hide()
-        Linking.removeEventListener('url', urlHandler)
+        if (this.urlEventListener) {
+          this.urlEventListener.remove()
+          this.urlEventListener = null
+        }
         resolve(event.url)
       }
-      Linking.addEventListener('url', urlHandler)
+      this.urlEventListener = Linking.addEventListener('url', urlHandler)
       if (!triggerError) {
         Authcore.showNormalUrl(url, closeOnLoad)
       } else {
         Authcore.showAuthUrl(url, closeOnLoad, (error, redirectURL) => {
-          Linking.removeEventListener('url', urlHandler)
+          if (this.urlEventListener) {
+            this.urlEventListener.remove()
+            this.urlEventListener = null
+          }
```

Comment:
If `Linking` is removed, should remove import too?

Reply:
We need it for `Linking.addEventListener`